### PR TITLE
[VEN-1117] Governance New Repo

### PR DESCRIPTION
**Pull request by UrAvgDeveloper**
This PR is new repo of governance and it's test

- contracts
- tests
- linting
- CI workflow

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,10 @@
+INFURA_API_KEY="zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz"
+MNEMONIC="here is where your twelve words mnemonic should be put my friend"
+
+# Block explorer API keys
+ARBISCAN_API_KEY="zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz"
+BSCSCAN_API_KEY="zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz"
+ETHERSCAN_API_KEY="zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz"
+OPTIMISM_API_KEY="zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz"
+POLYGONSCAN_API_KEY="zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz"
+SNOWTRACE_API_KEY="zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz"
```
Comment (chechu) [inline .env.example:10]:
```suggestion
# Block explorer API keys
BSCSCAN_API_KEY="zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz"
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+<!--
+  Any non-WIP PR should have all the checkmarks set.
+  If a checkmark is not applicable to your PR, mark it as done
+-->
+
+- [ ] I have updated the documentation to account for the changes in the code.
+- [ ] If I added new functionality, I added tests covering it.
+- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
+- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).
```
Comment (chechu) [inline .github/pull_request_template.md:17]: I don't see the `docs/CONTRIBUTING.md` file in the PR

File: .github/workflows/cd.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      - master
+      - develop
+
+jobs:
+  release:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (chechu) [inline .github/workflows/cd.yaml:15]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/cd.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - master
```
Comment (chechu) [inline .github/workflows/cd.yaml:6]:
```suggestion
      - main
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      - main
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
```
Comment (chechu) [inline .github/workflows/ci.yml:20]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+          echo "## Test results" >> $GITHUB_STEP_SUMMARY
+          echo "✅ Passed" >> $GITHUB_STEP_SUMMARY
+
+  docgen:
+    name: Docgen
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
```
Comment (chechu) [inline .github/workflows/ci.yml:66]:
```suggestion
        uses: actions/checkout@v3
```

File: .releaserc
Comment (chechu) [inline .releaserc:1]: Let's wait for merging https://github.com/VenusProtocol/venus-protocol-api/pull/329 first, after having an agreement on the different topics. And then we can replicate that configuration here

File: contracts/Governance/AccessControlled.sol [hunk 7]
```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.13;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
+
+import "./IAccessControlManager.sol";
+
+abstract contract AccessControlled is Initializable, Ownable2StepUpgradeable {
```
Comment (chechu) [inline contracts/Governance/AccessControlled.sol:9]: We would need a natspec comment here

File: contracts/Governance/GovernorBravoDelegate.sol [hunk 8]
```diff
@@ -0,0 +1,516 @@
+pragma solidity ^0.5.16;
```
Comment (chechu) [inline contracts/Governance/GovernorBravoDelegate.sol:1]: How are we going to deal with contracts using different solidity versions?

File: package.json [hunk 9]
```diff
@@ -0,0 +1,99 @@
+{
+  "name": "@venusprotocol/governance-contracts",
+  "description": "",
+  "version": "1.0.0",
```
Comment (chechu) [inline package.json:4]: I would set 0.0.1, leaving semantic-release to release the first version

File: package.json [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+    "lint:sol:fix": "prettier --write \"contracts/**/*.sol\"",
+    "prettier": "prettier --config ./.prettierrc.yaml --write \"**/*.{js,json,md,ts,yaml,yml,sol}\"",
+    "prettier:check": "prettier --check --config ./.prettierrc.yaml \"**/*.{js,json,md,ts,yaml,yml,sol}\"",
+    "docgen": "hardhat docgen"
+  },
+  "dependencies": {
+    "@openzeppelin/contracts": "^4.8.2",
+    "@openzeppelin/contracts-upgradeable": "^4.8.2",
+    "bignumber.js": "^9.1.1",
+    "semantic-release": "^19.0.3"
```
Comment (chechu) [inline package.json:97]: This would be a dev-dependency

**Review by chechu — COMMENTED** [review]
Would it make sense to include (as a dev dependency) the contracts from the core repo, to reuse the following contracts: XVS, XVSStore, XVSVault, XVSVaultErrorReporter, XVSVaultProxy, XVSVaultStorage. AFAIU we are not modifying those contracts in this repo, we are just using it in the tests.

This way we would avoid duplicated code in the GH organisation

File: contracts/Governance/GovernorBravoDelegate.sol (hunk 8 above)
Comment (aditya-routv) [reply contracts/Governance/GovernorBravoDelegate.sol:1]: We will create multiple Solidity compilers in the configuration to address the scenario where these lower version contracts are currently not interacting with any higher version contracts.

**Review by defcon022 — APPROVED** [review]
LGTM!

File: venusprotocol-governance-contracts-0.0.1.tgz
Comment (chechu) [inline venusprotocol-governance-contracts-0.0.1.tgz:1]: This file shouldn't be finally included. I suppose we have it here now for sharing it easily, but let's remove it before merging the PR

**Pull request merged**